modules/javafx.graphics/src/main/java/javafx/css/StyleableBooleanProperty.java

```diff
@@ -65,8 +69,15 @@ public StyleableBooleanProperty(boolean initialValue) {
     /** {@inheritDoc} */
     @Override
     public void applyStyle(StyleOrigin origin, Boolean v) {
-        // call set here in case it has been overridden in the javafx.beans.property
-        set(v.booleanValue());
+        TransitionDefinition transition = this.origin != null && getBean() instanceof Node node ?
+            NodeHelper.findTransitionDefinition(node, getCssMetaData()) : null;
+
+        if (transition != null) {
+            timer = TransitionTimer.run(new TransitionTimerImpl(this, v), transition);
+        } else {
+            setValue(v);
+        }
+
```

Comment:
I'm not sure how I feel about this; the changes made don't really seem to belong here.

The `StyleableXXXProperty` classes are convenient helpers, but all that matters is that they are `StyleableProperty` implementations.  There is no requirement to use the helpers.  So what happens when a property defines their own helper, or implements `StyleableProperty` directly?  Or if classes are refactored at some point and they decide to stop using these helpers?

Reply:
I don't see how I can make CSS transitions work without some property-specific implementation details. You're right that there is no requirement to use these classes. An easy solution would be to just document that CSS transitions are only supported for `StyleableXYZProperty` implementations.